### PR TITLE
Removes Warrior Shield, Implements Lifesteal and Cooldown on Slash Reductions.

### PR DESCRIPTION
**Pull request by usnpeepoo**
## About The Pull Request

Warrior slash shield has been removed, in place for it is lifesteal from slashes and a slight cooldown reduction per slash. Warrior also gains 50 more health. The lifesteal is comparable to shield, but requires 4 slashes to gain. The %ge of health gained increases with more marines near the warrior by 3 tiles up to a maximum of 9 (2 marines = 2% more lifesteal). This increased health shows a glow, and a cue in chat when it happens. Warrior also gains a very slight cooldown reduction per slash (0.30 seconds worth). 
Huge thanks to Nivrak for allowing me to reuse their lifesteal PR, and helping me with the visual cue for it.

## Why It's Good For The Game

Warriors shield is unhealthy for current CM gameplay, and this PR hopes to fix it through something similar to berserkers system of life steal. With shields, warriors have an invisible (to marines) extra health bar, that is permanent and always recoupable. By removing it, and implementing life steal and slash cooldown reduction warriors no longer have the points mentioned above, rather they are rewarded more intuitively for hitting abilities, finding opportunities to hit their slashes a lot more, and being rewarded for doing so without giving them a shield. By allowing warrior to heal and acquire stronger heals by being more risky with their abilities, and also giving them less cooldowns to initiate fights with per slash, warriors are now able to fit the role of a frontline-ambusher caste that can continue fighting through the use of its abilities while dealing with the issue with shields being in that role.

This PR also aims to be in line with the xeno shield design doc, and seeks to bring one of the goals to light by making warrior buff itself in a different way than simply shields. 
You can read the design doc here, https://github.com/cmss13-devs/cmss13-design-docs/blob/main/xeno/shields.md.

## Changelog

:cl:
balance: Removed shielding slash from warrior. Warrior slashes now heal for (7 + 1 per enemy within 3 tiles, up to 2)% of missing health with a minimum of 17 and a maximum of 40. There is now a visual, and chat cue for when a warrior gains extra health per marine.
balance: Warriors health is reverted back to 500, up from 450.
balance: Warriors slash now reduces cooldowns from all abilities by 0.30 seconds worth. 
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/Warrior.dm [hunk 1]
```diff
@@ -126,29 +126,53 @@
...
+	if (!cAction3.action_cooldown_check())
+		cAction3.reduce_cooldown(slash_charge_cdr)
+
+/datum/behavior_delegate/warrior_base/melee_attack_additional_effects_target(mob/living/carbon/A)
+	..()
+
+	var/final_lifesteal = lifesteal_percent
+	var/list/mobs_in_range = oviewers(lifesteal_range, bound_xeno)
+
+	for(var/mob/M as anything in mobs_in_range)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/castes/Warrior.dm:157]: more detailed var name for M

File: code/modules/mob/living/carbon/xenomorph/castes/Warrior.dm [hunk 2]
```diff
@@ -126,29 +126,53 @@
...
+
+		if(bound_xeno.can_not_harm(M))
+			continue
+
+		final_lifesteal++
+
+// This part is then outside the for loop
+		if(final_lifesteal >= max_lifesteal)
+			bound_xeno.add_filter("empower_rage", 1, list("type" = "outline", "color" = color, "size" = 1))
+			bound_xeno.visible_message(SPAN_DANGER("[bound_xeno.name]'s glows as it heals even more from its injuries!."), SPAN_XENODANGER("You glow as your heal even more from your injuries!"))
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/castes/Warrior.dm:172]:
```suggestion
			bound_xeno.visible_message(SPAN_DANGER("[bound_xeno.name] glows as it heals even more from its injuries!."), SPAN_XENODANGER("You glow as you heal even more from your injuries!"))
```

**Comment by Carnage068** [comment]
Holy fucking shit, this actually sounds like a really good idea. To me, at least. It sounds like it would push Warrior further in the direction of a frontline ambusher, instead of just an ambusher, and having higher risk + reward always sounds like a plus in my book. It's also just a lot more interesting and nuanced than a flat shield is.

**Comment by Carnage068** [comment]
FUCK YEAH

**Comment by PhantomEpicness** [comment]
W pr

File: code/modules/mob/living/carbon/xenomorph/abilities/warrior/warrior_abilities.dm [hunk 3]
```diff
@@ -5,7 +5,8 @@
 	ability_name = "toggle agility"
 	macro_path = /datum/action/xeno_action/verb/verb_toggle_agility
 	action_type = XENO_ACTION_CLICK
-	xeno_cooldown = 10
+	ability_primacy = XENO_PRIMARY_ACTION_4
+	xeno_cooldown = 40
 
 /datum/action/xeno_action/onclick/toggle_agility/can_use_action()
 	var/mob/living/carbon/Xenomorph/X = owner
```
Comment (Shad0vvs) [inline code/modules/mob/living/carbon/xenomorph/abilities/warrior/warrior_abilities.dm:12]:
```suggestion
```

**Review by Shad0vvs — CHANGES_REQUESTED** [review]
### Possible balance/consistency concerns to consider @ balance reviewer:
It appears the lifesteal regen can be done while the xeno is on fire, this is inconsistent with any other heal ability.
Agility state check is currently set to permissive. (Probably not a big deal)

### Requested changes.
Add `/datum/action/xeno_action/onclick/toggle_agility` to base actions line ~50 on Warrior.dm so we have it in our top left verbs.
Remove all of can_use_action() in warrior_abilities.dm ~line 10, the checks are redundant.
(You probably also need to rebase since I can punch from 2 tiles away)
Maybe add a message for if(agility) on the other abilities to_chat saying that you can't do this while on all fours or something like that.

**Comment by Carnage068** [comment]
Hold up since when the hell was it getting Agility back? Not that I have a problem with it of course, bringing back old content is great.

**Review by Shad0vvs — APPROVED** [review]

**Comment by Weuyn** [comment]
I think the green glow is too much. What was wrong with tge little vibrate they did when they got the shield? That was fine imo.

**Comment by usnpeepoo** [comment]
> I think the green glow is too much. They just end up green all the time and it looks weird. Maybe just a jiggle like the do atm when they get the shield but only if they get a lot of health back? Not if its just 1 or 2 marines.

it is currently the old shield slash jitter, and the glow has been reduced by a lot and the warden heal icon has been added onto it, so when a warrior gets extra lifesteal it's still visible

**Comment by usnpeepoo** [comment]
i shouldve searched up a how to rebase tutorial

**Comment by usnpeepoo** [comment]
squash commits (it squashed every merged pr for the last 2 weeks into this PR)

never again

**Pull request closed without merging**